plugin/storage/es/spanstore/reader.go

```diff
@@ -91,27 +91,30 @@ type SpanReader struct {
 	// this will be rounded down to UTC 00:00 of that day.
 	maxLookback             time.Duration
 	serviceOperationStorage *ServiceOperationStorage
+
+	timeNow func() time.Time
 }
 
 // NewSpanReader returns a new SpanReader with a metrics.
 func NewSpanReader(client es.Client, logger *zap.Logger, maxLookback time.Duration, metricsFactory metrics.Factory) spanstore.Reader {
-	return storageMetrics.NewReadMetricsDecorator(newSpanReader(client, logger, maxLookback), metricsFactory)
+	return storageMetrics.NewReadMetricsDecorator(newSpanReader(client, logger, maxLookback, time.Now), metricsFactory)
```

Comment:
please break to multiple lines, at least for the top-level function (newSpanReader can stay on one line)